mbs_results/staging/back_data.py

```diff
@@ -96,12 +96,26 @@ def read_back_data(config: dict) -> pd.DataFrame:
         qv_df, cp_df, how="left", on=[config["period"], config["reference"]]
     )
 
-    back_data_all_cols = pd.merge(
-        qv_and_cp, finalsel, how="left", on=[config["period"], config["reference"]]
+    finalsel[config["period"]] = convert_column_to_datetime(finalsel[config["period"]])
+
+    qv_and_cp[config["period"]] = convert_column_to_datetime(
+        qv_and_cp[config["period"]]
     )
 
-    back_data_all_cols[config["period"]] = convert_column_to_datetime(
-        back_data_all_cols[config["period"]]
+    qv_and_cp_period = qv_and_cp[config["period"]].unique()
+    finalsel_period = finalsel[config["period"]].unique()
+
```

Comment:
I think because this is a one time use bit of code, it makes sense to add a dummy variable in for the join.
That way it can be removed easily in the future once its no longer needed / once we move away from csw 
```suggestion
join_type = "left"
```